hail/python/hailtop/batch/batch_pool_executor.py

```diff
@@ -132,7 +131,14 @@ def __init__(self, *,
         self.directory = self.backend.remote_tmpdir + f'batch-pool-executor/{self.name}/'
         self.inputs = self.directory + 'inputs/'
         self.outputs = self.directory + 'outputs/'
-        self.fs = RouterAsyncFS(gcs_kwargs={'gcs_requester_pays_configuration': project})
+        if project is not None:
+            warnings.warn(
+                'The project parameter of BatchPoolExecutor is deprecated. Please '
+                'use the gcs_requester_pays_project parameter of ServiceBackend.'
+            )
+            self._fs = RouterAsyncFS(gcs_kwargs={'gcs_requester_pays_configuration': project})
+        else:
+            self._fs = None
```

Comment:
This just seemed like old cruft. If there's no project, we should just use the backend's configuration and FS.